### PR TITLE
Implement rewind and sequence with test cases

### PR DESCRIPTION
**Pull request by itancio**

### CODE REVIEW

**Review by ggorlen — COMMENTED** [review]
I haven't had a chance to do extensive testing, just dropping a comment for now. Feel free to merge if you're blocked on this and feel pretty confident in it. I may not be able to get to it until Sunday.

**Review by Nireves333 — COMMENTED** [review]
I haven't read through the code but from basic testing, I am running into some problems. When using the arrow keys to move and clicking the undo button or 'z', it resets the whole board instead of undoing one move. But it works normally when using BFS.

File: src/sokoban.cpp [hunk 1]
```diff
@@ -249,6 +247,37 @@ void Sokoban::change_level(unsigned int level_number) {
...
+            break;
+        }
+    }
+
+    return true;
+}
+
+std::string Sokoban::sequence() {
+    std::string sequence = "";
+    for (const auto& direction : moves) {
```
[the remark this answers is not in the document]
Comment (itancio) [reply src/sokoban.cpp:279]: I don't understand. I am using push_back to add each character to a string. push_back has constant time O(1). The time complexity that I am converting each direction to a string is basically the length of moves. Am I understanding this correctly?

**Comment by itancio** [comment]
Good catch. Thank you. I'll make adjustments.

File: src/sokoban.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ggorlen) [reply src/sokoban.cpp:279]: I see strings are mutable in C++, so `+=` is O(1) based on [this SO answer](https://stackoverflow.com/a/15400811/6243352). Wrong language!

**Comment by Nireves333** [comment]
Awesome! I think the code looks good. The only other issue I noticed is that the console repeatedly prints "solved" when the player is moved once the level is complete, but I believe that could be easily fixed in the future once we have a "level complete" screen.

**Pull request merged**